gammapy/maps/map_dict.py

```diff
@@ -0,0 +1,139 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from collections import UserDict
+from collections.abc import MutableMapping
+from astropy.io import fits
+from gammapy.maps import Map
+from gammapy.utils.scripts import make_path
+
+__all__ = ["Maps"]
+
+
+class Maps(MutableMapping):
+    """A Dictionary containing Map objects sharing the same geometry.
+
+    This class simplifies handling and I/O of maps collections.
+
+    For maps with different geometries, use a regular dict.
+    """
+
+    def __init__(self, **kwargs):
+        self._geom = None
+        self._data = {}
+        for key, value in kwargs.items():
+            self.__setitem__(key, value)
+
+    @property
+    def geom(self):
+        """Map geometry (`Geom`)"""
+        return self._geom
+
+    def __setitem__(self, key, value):
+        if value is not None and not isinstance(value, Map):
+            raise ValueError(f"MapDict can only contain Map objects.")
```

Comment:
Add the actual and expected type here, maybe like  `f"Maps can only contain Map objects, got {type(value)} instead."`

Reply:
Done